.github/workflows/main.yml

```diff
@@ -266,19 +266,13 @@ jobs:
           python-version: '3.10'
           architecture: x64
 
-      - uses: actions/download-artifact@v3
-        with:
-          name: wheelhouse
-          path: wheelhouse
+      - uses: actions/download-artifact@v4
 
-      - uses: actions/download-artifact@v3
-        with:
-          name: version
-          path: .
+      - name: Combine wheelhouses
+        run: mv wheelhouse-*/* wheelhouse
```

Comment:
there was a breaking change in both upload-artifact@v4 and download-artifact@v4 where it is now not allowed anymore to upload multiple times to the same artifact name (which up til @v3 would result in creating a new subdirectory in the object⁠).



i worked around this by creating a wheelhouse-<platform> per parallel job (matrix over platform) and then downloading all artifacts and moving all downloaded artifacts in wheelhouse-* directories into the wheelhouse directory. this yields the same output when listing the assets: https://github.com/PowerGridModel/power-grid-model/actions/runs/7223577211/job/19684378764#step:6:1


```
​Run ls ./wheelhouse/ -al
total 5120
drwxr-xr-x  2 runner docker    4096 Dec 15 15:35 .
drwxr-xr-x 16 runner docker    4096 Dec 15 15:35 ..
-rw-r--r--  1 runner docker  727885 Dec 15 15:35 power-grid-model-1.6.39.tar.gz
-rw-r--r--  1 runner docker  559533 Dec 15 15:35 power_grid_model-1.6.39-py3-none-macosx_10_9_x86_64.whl
-rw-r--r--  1 runner docker  521874 Dec 15 15:35 power_grid_model-1.6.39-py3-none-macosx_11_0_arm64.whl
-rw-r--r--  1 runner docker  655115 Dec 15 15:35 power_grid_model-1.6.39-py3-none-manylinux_2_24_aarch64.whl
-rw-r--r--  1 runner docker  710367 Dec 15 15:35 power_grid_model-1.6.39-py3-none-manylinux_2_24_x86_64.whl
-rw-r--r--  1 runner docker 1567670 Dec 15 15:35 power_grid_model-1.6.39-py3-none-musllinux_1_2_x86_64.whl
-rw-r--r--  1 runner docker  481711 Dec 15 15:35 power_grid_model-1.6.39-py3-none-win_amd64.whl
```